geopandas/array.py

```diff
@@ -64,6 +64,8 @@ def _isna(value):
         return True
     elif isinstance(value, float) and np.isnan(value):
         return True
+    elif compat.PANDAS_GE_10 and isinstance(value, type(pd.NA)):
```

Comment:
@martinfleis using type-checking of `pd.NA` doesn't seem quite right here.

I'm not familiar with the decisions to check manually NA here, but `pd.isnull(value)` seems to do the right thing for `pd.isnull(None)`, `pd.isnull(pd.NA)`, and `pd.isnull(np.nan)`.  Is there a reason for not using `pd.isnull()` here?

Note: [`pd.isna()`](https://github.com/pandas-dev/pandas/blob/1.0.x/pandas/core/dtypes/missing.py#L49) may work here too, `pd.isnull()`  is an alias to it.

Reply:
> Is there a reason for not using `pd.isnull()` here?

I assume it has something to do with array interface of shapely geometries. There might be cases when pandas returns array of boolean values (not sure when, I am not able to get such a result now). @jorisvandenbossche ?

> using type-checking of pd.NA doesn't seem quite right here.

The way how it is done or the fact that we check for type? I am not importing the type itself because it would come from a private part of pandas (`from pandas._libs.missing import NA`).

Reply:
The reason we have this "special" version is to ensure it only works on scalars (it is noted in the docstring, which is hidden in the diff here, but if you expand the diff a bit upwards you can see it)

Reply:
Got it, thanks for the clarification.  

So is the reason to do type checking because `value is pd.NA` doesn't work as it should?  It seems to work in my manual testing, but I must be overlooking something.  If type-checking is the right way to do this, then go for it.

Side note: it would be good to add `pd.NA` to the docstring list of na types.

Reply:
You are right, I am an idiot. `value == pd.NA` returns `NA` and I did not try `value is pd.NA`... Fixed, including the docstring.